### PR TITLE
[prometheus-blackbox-exporter] define metricRelabelings in selfservicemonitor.yaml to handle additionalMetricsRelabels

### LINKED ISSUE #4296: [prometheus-blackbox-exporter] additionalMetricsRelabels in the values file, but not handled in selfservicemonitor
### Describe the bug a clear and concise description of what the bug is.

additionalMetricsRelabels is a propertie in the [values.yaml](https://github.com/prometheus-community/helm-charts/blob/b4001815202cc773a51ad9b9de5a56eca18660d6/charts/prometheus-blackbox-exporter/values.yaml) file but not in the [selfservicemonitor.yaml file](https://github.com/prometheus-community/helm-charts/blob/b4001815202cc773a51ad9b9de5a56eca18660d6/charts/prometheus-blackbox-exporter/templates/selfservicemonitor.yaml). resulting in the fact that nothing hapens with the value.

Thanks for checking.

### What's your helm version?

8.11.0

### What's your kubectl version?

latest

### Which chart?

prometheus-blackbox-exporter

### What's the chart version?

8.11.0

### What happened?

_No response_

### What you expected to happen?

_No response_

### How to reproduce it?

_No response_

### Enter the changed values of values.yaml?

_No response_

### Enter the command that you execute and failing/misfunctioning.

serviceMonitor:
  ## If true, a ServiceMonitor CRD is created for a prometheus operator
  ## https://github.com/coreos/prometheus-operator for blackbox-exporter itself
  ##
  selfMonitor:
    enabled: true
    additionalMetricsRelabels:
    - action: lowercase
      sourceLabels:
        - team
      targetLabel: team
    additionalRelabeling: []
    labels:
      prometheus: infra
    path: /metrics
    interval: 1m
    scrapeTimeout: 30s

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Sheikh-Abubaker**
#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #4296 

##   Generated ServiceMonitor resource after defining metricRelabelings in selfservicemonitor.yaml, and when additionalMetricsRelabels set to an empty list.
```
helm template . -f override-values.yaml

---
# Source: prometheus-blackbox-exporter/templates/selfservicemonitor.yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: release-name-prometheus-blackbox-exporter
  namespace: default
  labels:
    helm.sh/chart: prometheus-blackbox-exporter-8.13.0
    app.kubernetes.io/name: prometheus-blackbox-exporter
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "v0.24.0"
    app.kubernetes.io/managed-by: Helm
spec:
  endpoints:
  - path: /metrics
    interval: 30s
    scrapeTimeout: 30s
    scheme: http
    metricRelabelings:
      - action: keep
        regex: prometheus
        sourceLabels:
        - __meta_kubernetes_pod_label_team
      - action: drop
        regex: west_europe
        sourceLabels:
        - __meta_kubernetes_pod_label_datacenter
  jobLabel: "release-name"
  selector:
    matchLabels:
      app.kubernetes.io/name: prometheus-blackbox-exporter
      app.kubernetes.io/instance: release-name
  namespaceSelector:
    matchNames:
      - default

```
## additionalMetricsRelabels changed to empty list due to this warning
```
helm template . -f override-values.yaml
 coalesce.go:286: warning: cannot overwrite table with non table for prometheus-blackbox-exporter.serviceMonitor.selfMonitor.additionalMetricsRelabels (map[])
 ```
  ```
  # override-values.yaml 

serviceMonitor:
  ## If true, a ServiceMonitor CRD is created for a prometheus operator
  ## https://github.com/coreos/prometheus-operator for blackbox-exporter itself
  ##
  selfMonitor:
    enabled: true
    additionalMetricsRelabels:
    - sourceLabels:
      - __meta_kubernetes_pod_label_team
      regex: "prometheus"
      action: keep
    - sourceLabels:
      - __meta_kubernetes_pod_label_datacenter
      regex: west_europe
      action: drop
    additionalRelabeling: []
    labels: {}
    path: /metrics
    interval: 30s
    scrapeTimeout: 30s
   ```

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-blackbox-exporter/values.yaml [hunk 1]
```diff
@@ -271,7 +271,7 @@ serviceMonitor:
   ##
   selfMonitor:
     enabled: false
-    additionalMetricsRelabels: {}
+    additionalMetricsRelabels: []
```
Comment (monotek) [inline charts/prometheus-blackbox-exporter/values.yaml:274]: thsi will break: https://github.com/prometheus-community/helm-charts/blob/e8b1efd4c1709a2c878d4e9923202ec303a897dc/charts/prometheus-blackbox-exporter/templates/selfpodmonitoring.yaml#L21

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-blackbox-exporter/values.yaml (hunk 1 above)
Comment (Sheikh-Abubaker) [reply charts/prometheus-blackbox-exporter/values.yaml:274]: @monotek can you please elaborate?

Comment (monotek) [reply charts/prometheus-blackbox-exporter/values.yaml:274]: you switch from list to map but map is used elsewhere.

Comment (Sheikh-Abubaker) [reply charts/prometheus-blackbox-exporter/values.yaml:274]: > but map is used elsewhere.

*list right ?

Comment (Sheikh-Abubaker) [reply charts/prometheus-blackbox-exporter/values.yaml:274]: > you switch from list to map but map is used elsewhere.

what do you suggest should I revert it to map again ?

Comment (Sheikh-Abubaker) [reply charts/prometheus-blackbox-exporter/values.yaml]: please checkout the latest commit I've reverted `additionalMetricsRelabels` to map.

**Review by monotek — APPROVED** [review]

**Pull request merged**